src/newsimplification/Simplifier.ts

```diff
@@ -0,0 +1,150 @@
+   * 
+   * @param model to simplify to simplify schema into 
+   * @param schema to simplify
+   */
+  private simplifyModel(model: CommonModel, schema: Schema) {
+    if (schema.type !== undefined) {
+      model.type = schema.type;
+    }
+
+    //All schemas of type object MUST have ids
+    if (model.type !== undefined && model.type.includes('object')) {
+      model.$id = simplifyName(schema) || `anonymSchema${this.anonymCounter++}`;
+    } else if (schema.$id !== undefined) {
+      model.$id = simplifyName(schema);
+    }
+
+    if (typeof schema !== 'boolean' && schema.required !== undefined) {
+      model.required = schema.required;
+    }
+
+    simplifyProperties(schema, model, this);
+    simplifyAllOf(schema, model, this);
+
+    this.combineSchemas(schema.oneOf, model, schema);
+    this.combineSchemas(schema.anyOf, model, schema);
+    this.combineSchemas(schema.then, model, schema);
+    this.combineSchemas(schema.else, model, schema);
+  }
+
+  /**
+   * Go through schema(s) and combine the simplified models together.
+   * 
+   * @param schema to go through
+   * @param currentModel the current output
+   */
+  combineSchemas(schema: (Schema | boolean) | (Schema | boolean)[] | undefined, currentModel: CommonModel, rootSchema: Schema) {
+    if (typeof schema !== 'object') return;
+    if (Array.isArray(schema)) {
+      schema.forEach((forEachSchema) => {
+        this.combineSchemas(forEachSchema, currentModel, rootSchema);
+      });
+    } else {
+      const models = this.simplify(schema, false);
+      if (models.length > 0) {
+        CommonModel.mergeCommonModels(currentModel, models[0], rootSchema);
+      }
+    }
+  }
+
+  /**
+  * This function splits up a model if needed and add the new model to the list of models.
+  * 
+  * @param model check if it should be split up
+  * @param models which have already been split up
+  */
+  private splitModels(model: CommonModel): CommonModel {
+    if (isModelObject(model)) {
+      const switchRootModel = new CommonModel();
+      switchRootModel.$ref = model.$id;
+      this.iteratedModels[`${model.$id}`] = model;
+      return switchRootModel;
+    }
+    return model;
+  }
+
+  /**
+   * Split up all models which should and use ref instead.
+   * 
+   * @param model to ensure are split
+   * @param models which are already split
+   */
+  ensureModelsAreSplit(model: CommonModel) {
+    // eslint-disable-next-line sonarjs/no-collapsible-if
+    if (model.properties) {
+      const existingProperties = model.properties;
+      for (const [prop, propSchema] of Object.entries(existingProperties)) {
+        model.properties[`${prop}`] = this.splitModels(propSchema);
+      }
+    }
+  }
+}
+
+/**
+ * This is the default wrapper for the simplifier class which always create a new instance of the simplifier. 
+ * 
+ * @param schema to simplify
+ */
+export function simplify(schema: Schema | boolean, options?: SimplificationOptions): CommonModel[] {
```

Comment:
I guess that it is too much for encapsulation. I am not sure but I think that some of people would say that you violeted KISS rule here. Now you have multiple ways to actually use simplifier.